blueflood-http/src/main/java/com/rackspacecloud/blueflood/inputs/handlers/HttpMetricsIngestionHandler.java

```diff
@@ -99,19 +99,32 @@ public void handle(ChannelHandlerContext ctx, HttpRequest request) {
 
             requestCount.inc();
             final String tenantId = request.getHeader("tenantId");
-            JSONMetricsContainer jsonMetricsContainer = null;
+            JSONMetricsContainer jsonMetricsContainer;
+            List<Metric> metrics;
+
             final Timer.Context jsonTimerContext = jsonTimer.time();
 
             final String body = request.getContent().toString(Constants.DEFAULT_CHARSET);
             try {
                 jsonMetricsContainer = createContainer(body, tenantId);
 
                 if (jsonMetricsContainer == null) {
-                    log.warn(ctx.getChannel().getRemoteAddress() + " No valid metrics");
+                    log.warn(ctx.getChannel().getRemoteAddress() + " No valid json for container.");
                     DefaultHandler.sendResponse(ctx, request, "No valid metrics", HttpResponseStatus.BAD_REQUEST);
                     return;
+                } else if (jsonMetricsContainer.getJsonMetrics().isEmpty()) {
+                    log.warn(ctx.getChannel().getRemoteAddress() + " No valid json metrics to parse.");
+                    DefaultHandler.sendResponse(ctx, request, "Error converting JSON payload to metric objects", HttpResponseStatus.BAD_REQUEST);
+                    return;
```

Comment:
According to the current tests, for regular metrics if they valid but lame json, e.g. '[]' or '{}', it should return a 400 error code with that error message.

But for aggregated metrics, if they send empty array, then it's 200.  Do you want me to modify it so that they're both consistent one way or the other?